src/components/WordSearch/WordSearch.js

```diff
@@ -0,0 +1,250 @@
+/* eslint-disable array-callback-return */
+import React, {useState, useEffect} from 'react';
+import { BackgroundImg } from '../../pages/SignIn/SignInElements';
+import img from "../../images/Login_Signup.png"
+import { Line } from '../../pages/SignUp/SignUpElements';
+import HomeIcon from '@mui/icons-material/Home';
+import IconButton from '@mui/material/IconButton';
+import { useNavigate } from 'react-router-dom';
+import { QuestionContainer, Rectangle, Board, QuestionTypeText, HowToPlayText, WordBank, Word } from './WordSearchElements';
+import Letter from './Letter';
+import SuccessPopup from '../SuccessPopup/SuccessPopup';
+import axios from 'axios';
+
+function WordSearch(props) {
+
+    const [wordsFound, updateWordsFound] = useState([]);
+    const [wordSelected, setWordSelected] = useState("");
+    const [letterSelectedCoordinate, setLetterSelectedCoordinate] = useState({rowIndex: -1, colIndex: -1});
+    const [direction, setDirection] = useState("");
+    const [lettersInWordSelected, updateLettersInWordSelected] = useState([]);
+    const [lettersDisabled, updateLettersDisabled] = useState([]);
+    const navigate = useNavigate();
+
+    useEffect(() => {
```

Comment:
I may not understand useeffect correctly but wouldn't the page need to be reloaded for this to be performed? If so the wordSearchCompleted may not be called

Reply:
useEffect is ran every time the page is reloaded only and every time the variable in its dependency array changes (for instance [wordSelected]). So wordSearchCompleted is called only when the word search has been successfully completed.